File: samples/snippets/csharp/VS_Snippets_CLR/string.split3/CS/omit.cs
﻿using System;

class Sample
{
    public static void Main()
    {
    //<snippet1>
    // This example demonstrates the String() methods that use
    // the StringSplitOptions enumeration.
    string s1 = ",ONE,,TWO,,,THREE,,";
    string s2 = "[stop]" +
                "ONE[stop][stop]" +
                "TWO[stop][stop][stop]" +
                "THREE[stop][stop]";
    char[] charSeparators = new char[] {','};
    string[] stringSeparators = new string[] {"[stop]"};
    string[] result;
    // ------------------------------------------------------------------------------
    // Split a string delimited by characters.
    // ------------------------------------------------------------------------------
    Console.WriteLine("1) Split a string delimited by characters:\n");

    // Display the original string and delimiter characters.
    Console.WriteLine("1a )The original string is \"{0}\".", s1);
    Console.WriteLine("The delimiter character is '{0}'.\n",
                       charSeparators[0]);

    // Split a string delimited by characters and return all elements.
    Console.WriteLine("1b) Split a string delimited by characters and " +
                      "return all elements:");
    result = s1.Split(charSeparators, StringSplitOptions.None);
    Show(result);

    // Split a string delimited by characters and return all non-empty elements.
    Console.WriteLine("1c) Split a string delimited by characters and " +
                      "return all non-empty elements:");
    result = s1.Split(charSeparators, StringSplitOptions.RemoveEmptyEntries);
    Show(result);

    // Split the original string into the string and empty string before the
    // delimiter and the remainder of the original string after the delimiter.
    Console.WriteLine("1d) Split a string delimited by characters and " +
                      "return 2 elements:");
    result = s1.Split(charSeparators, 2, StringSplitOptions.None);
    Show(result);

    // Split the original string into the string after the delimiter and the
    // remainder of the original string after the delimiter.
    Console.WriteLine("1e) Split a string delimited by characters and " +
                      "return 2 non-empty elements:");
    result = s1.Split(charSeparators, 2, StringSplitOptions.RemoveEmptyEntries);
    Show(result);

    // ------------------------------------------------------------------------------
    // Split a string delimited by another string.
    // ------------------------------------------------------------------------------
    Console.WriteLine("2) Split a string delimited by another string:\n");

    // Display the original string and delimiter string.
    Console.WriteLine("2a) The original string is \"{0}\".", s2);
    Console.WriteLine("The delimiter string is \"{0}\".\n", stringSeparators[0]);

    // Split a string delimited by another string and return all elements.
    Console.WriteLine("2b) Split a string delimited by another string and " +
                      "return all elements:");
    result = s2.Split(stringSeparators, StringSplitOptions.None);
    Show(result);

    // Split the original string at the delimiter and return all non-empty elements.
    Console.WriteLine("2c) Split a string delimited by another string and " +
                      "return all non-empty elements:");
    result = s2.Split(stringSeparators, StringSplitOptions.RemoveEmptyEntries);
    Show(result);

    // Split the original string into the empty string before the
    // delimiter and the remainder of the original string after the delimiter.
    Console.WriteLine("2d) Split a string delimited by another string and " +
                      "return 2 elements:");
    result = s2.Split(stringSeparators, 2, StringSplitOptions.None);
    Show(result);

    // Split the original string into the string after the delimiter and the
    // remainder of the original string after the delimiter.
    Console.WriteLine("2e) Split a string delimited by another string and " +
                      "return 2 non-empty elements:");
    result = s2.Split(stringSeparators, 2, StringSplitOptions.RemoveEmptyEntries);
    Show(result);

    // Display the array of separated strings using a local function
    void Show(string[] entries)
    {
        Console.WriteLine("The return value contains these {0} elements:", entries.Length);
        foreach (string entry in entries)
        {
            Console.Write("<{0}>", entry);
        }
        Console.Write("\n\n");
    }

    /*
    This example produces the following results:

    1) Split a string delimited by characters:

    1a )The original string is ",ONE,,TWO,,,THREE,,".
    The delimiter character is ','.

    1b) Split a string delimited by characters and return all elements:
    The return value contains these 9 elements:
    <><ONE><><TWO><><><THREE><><>

    1c) Split a string delimited by characters and return all non-empty elements:
    The return value contains these 3 elements:
    <ONE><TWO><THREE>

    1d) Split a string delimited by characters and return 2 elements:
    The return value contains these 2 elements:
    <><ONE,,TWO,,,THREE,,>

    1e) Split a string delimited by characters and return 2 non-empty elements:
    The return value contains these 2 elements:
    <ONE><TWO,,,THREE,,>

    2) Split a string delimited by another string:

    2a) The original string is "[stop]ONE[stop][stop]TWO[stop][stop][stop]THREE[stop][stop]".
    The delimiter string is "[stop]".

    2b) Split a string delimited by another string and return all elements:
    The return value contains these 9 elements:
    <><ONE><><TWO><><><THREE><><>

    2c) Split a string delimited by another string and return all non-empty elements:
    The return value contains these 3 elements:
    <ONE><TWO><THREE>

    2d) Split a string delimited by another string and return 2 elements:
    The return value contains these 2 elements:
    <><ONE[stop][stop]TWO[stop][stop][stop]THREE[stop][stop]>

    2e) Split a string delimited by another string and return 2 non-empty elements:
    The return value contains these 2 elements:
    <ONE><TWO[stop][stop][stop]THREE[stop][stop]>

    */
    //</snippet1>
    }
}

File: snippets/visualbasic/System/string.split/omit.vb
' This example demonstrates the String() methods that use
' the StringSplitOptions enumeration.
Class Sample
    '<snippet1>
    Public Shared Sub StringSplitOptionsExamples()
        ' This example demonstrates the String.Split() methods that use
        ' the StringSplitOptions enumeration.

        ' Example 1: Split a string delimited by characters
        Console.WriteLine("1) Split a string delimited by characters:" & vbCrLf)

        Dim s1 As String = ",ONE,, TWO,, , THREE,,"
        Dim charSeparators() As Char = {","c}
        Dim result() As String

        Console.WriteLine("The original string is: ""{0}"".", s1)
        Console.WriteLine("The delimiter character is: '{0}'." & vbCrLf, charSeparators(0))

        ' Split the string and return all elements
        Console.WriteLine("1a) Return all elements:")
        result = s1.Split(charSeparators, StringSplitOptions.None)
        Show(result)

        ' Split the string and return all elements with whitespace trimmed
        Console.WriteLine("1b) Return all elements with whitespace trimmed:")
        result = s1.Split(charSeparators, StringSplitOptions.TrimEntries)
        Show(result)

        ' Split the string and return all non-empty elements
        Console.WriteLine("1c) Return all non-empty elements:")
        result = s1.Split(charSeparators, StringSplitOptions.RemoveEmptyEntries)
        Show(result)

        ' Split the string and return all non-whitespace elements with whitespace trimmed
        Console.WriteLine("1d) Return all non-whitespace elements with whitespace trimmed:")
        result = s1.Split(charSeparators, StringSplitOptions.RemoveEmptyEntries Or StringSplitOptions.TrimEntries)
        Show(result)


        ' Split the string into only two elements, keeping the remainder in the last match
        Console.WriteLine("1e) Split into only two elements:")
        result = s1.Split(charSeparators, 2, StringSplitOptions.None)
        Show(result)

        ' Split the string into only two elements with whitespace trimmed, keeping the remainder in the last match
        Console.WriteLine("1f) Split into only two elements with whitespace trimmed:")
        result = s1.Split(charSeparators, 2, StringSplitOptions.TrimEntries)
        Show(result)

        ' Split the string into only two non-empty elements, keeping the remainder in the last match
        Console.WriteLine("1g) Split into only two non-empty elements:")
        result = s1.Split(charSeparators, 2, StringSplitOptions.RemoveEmptyEntries)
        Show(result)

        ' Split the string into only two non-whitespace elements with whitespace trimmed, keeping the remainder in the last match
        Console.WriteLine("1h) Split into only two non-whitespace elements with whitespace trimmed:")
        result = s1.Split(charSeparators, 2, StringSplitOptions.RemoveEmptyEntries Or StringSplitOptions.TrimEntries)
        Show(result)


        ' Example 2: Split a string delimited by another string
        Console.WriteLine("2) Split a string delimited by another string:" & vbCrLf)

        Dim s2 As String = "[stop]" +
                    "ONE[stop] [stop]" +
                    "TWO  [stop][stop]  [stop]" +
                    "THREE[stop][stop]  "
        Dim stringSeparators() As String = {"[stop]"}


        Console.WriteLine("The original string is: ""{0}"".", s2)
        Console.WriteLine("The delimiter string is: ""{0}""." & vbCrLf, stringSeparators(0))

        ' Split the string and return all elements
        Console.WriteLine("2a) Return all elements:")
        result = s2.Split(stringSeparators, StringSplitOptions.None)
        Show(result)

        ' Split the string and return all elements with whitespace trimmed
        Console.WriteLine("2b) Return all elements with whitespace trimmed:")
        result = s2.Split(stringSeparators, StringSplitOptions.TrimEntries)
        Show(result)

        ' Split the string and return all non-empty elements
        Console.WriteLine("2c) Return all non-empty elements:")
        result = s2.Split(stringSeparators, StringSplitOptions.RemoveEmptyEntries)
        Show(result)

        ' Split the string and return all non-whitespace elements with whitespace trimmed
        Console.WriteLine("2d) Return all non-whitespace elements with whitespace trimmed:")
        result = s2.Split(stringSeparators, StringSplitOptions.RemoveEmptyEntries Or StringSplitOptions.TrimEntries)
        Show(result)


        ' Split the string into only two elements, keeping the remainder in the last match
        Console.WriteLine("2e) Split into only two elements:")
        result = s2.Split(stringSeparators, 2, StringSplitOptions.None)
        Show(result)

        ' Split the string into only two elements with whitespace trimmed, keeping the remainder in the last match
        Console.WriteLine("2f) Split into only two elements with whitespace trimmed:")
        result = s2.Split(stringSeparators, 2, StringSplitOptions.TrimEntries)
        Show(result)

        ' Split the string into only two non-empty elements, keeping the remainder in the last match
        Console.WriteLine("2g) Split into only two non-empty elements:")
        result = s2.Split(stringSeparators, 2, StringSplitOptions.RemoveEmptyEntries)
        Show(result)

        ' Split the string into only two non-whitespace elements with whitespace trimmed, keeping the remainder in the last match
        Console.WriteLine("2h) Split into only two non-whitespace elements with whitespace trimmed:")
        result = s2.Split(stringSeparators, 2, StringSplitOptions.RemoveEmptyEntries Or StringSplitOptions.TrimEntries)
        Show(result)

    End Sub

    ' Display the array of separated strings.
    Public Shared Sub Show(ByVal entries() As String)
        Console.WriteLine("The return value contains these {0} elements:", entries.Length)
        Dim entry As String
        For Each entry In entries
            Console.Write("<{0}>", entry)
        Next entry
        Console.Write(vbCrLf & vbCrLf)

    End Sub

    'This example produces the following results:
    '
    ' 1) Split a string delimited by characters:
    '
    ' The original string is: ",ONE,, TWO,, , THREE,,".
    ' The delimiter character is: ','.
    '
    ' 1a) Return all elements:
    ' The return value contains these 9 elements:
    ' <><ONE><>< TWO><>< >< THREE><><>
    '
    ' 1b) Return all elements with whitespace trimmed:
    ' The return value contains these 9 elements:
    ' <><ONE><><TWO><><><THREE><><>
    '
    ' 1c) Return all non-empty elements:
    ' The return value contains these 4 elements:
    ' <ONE>< TWO>< >< THREE>
    '
    ' 1d) Return all non-whitespace elements with whitespace trimmed:
    ' The return value contains these 3 elements:
    ' <ONE><TWO><THREE>
    '
    ' 1e) Split into only two elements:
    ' The return value contains these 2 elements:
    ' <><ONE,, TWO,, , THREE,,>
    '
    ' 1f) Split into only two elements with whitespace trimmed:
    ' The return value contains these 2 elements:
    ' <><ONE,, TWO,, , THREE,,>
    '
    ' 1g) Split into only two non-empty elements:
    ' The return value contains these 2 elements:
    ' <ONE>< TWO,, , THREE,,>
    '
    ' 1h) Split into only two non-whitespace elements with whitespace trimmed:
    ' The return value contains these 2 elements:
    ' <ONE><TWO,, , THREE,,>
    '
    ' 2) Split a string delimited by another string:
    '
    ' The original string is: "[stop]ONE[stop] [stop]TWO  [stop][stop]  [stop]THREE[stop][stop]  ".
    ' The delimiter string is: "[stop]".
    '
    ' 2a) Return all elements:
    ' The return value contains these 9 elements:
    ' <><ONE>< ><TWO  ><><  ><THREE><><  >
    '
    ' 2b) Return all elements with whitespace trimmed:
    ' The return value contains these 9 elements:
    ' <><ONE><><TWO><><><THREE><><>
    '
    ' 2c) Return all non-empty elements:
    ' The return value contains these 6 elements:
    ' <ONE>< ><TWO  ><  ><THREE><  >
    '
    ' 2d) Return all non-whitespace elements with whitespace trimmed:
    ' The return value contains these 3 elements:
    ' <ONE><TWO><THREE>
    '
    ' 2e) Split into only two elements:
    ' The return value contains these 2 elements:
    ' <><ONE[stop] [stop]TWO  [stop][stop]  [stop]THREE[stop][stop]  >
    '
    ' 2f) Split into only two elements with whitespace trimmed:
    ' The return value contains these 2 elements:
    ' <><ONE[stop] [stop]TWO  [stop][stop]  [stop]THREE[stop][stop]>
    '
    ' 2g) Split into only two non-empty elements:
    ' The return value contains these 2 elements:
    ' <ONE>< [stop]TWO  [stop][stop]  [stop]THREE[stop][stop]  >
    '
    ' 2h) Split into only two non-whitespace elements with whitespace trimmed:
    ' The return value contains these 2 elements:
    ' <ONE><TWO  [stop][stop]  [stop]THREE[stop][stop]>
    '
    '</snippet1>
End Class
